demos/2_Iterative/06_callbacks_how_to.ipynb

```diff
@@ -0,0 +1,683 @@
+    "In this example, we define a custom callback that saves the values of a wavelet-based estimator of Gaussian noise standard deviation, provided by `skimage`, at each iteration. Using TV regularization and a FISTA optimization algorithm with a lower bound, we compare how the noise level in the solution changes for two different regularization parameters: a small value (0.1) and a larger value (1).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import skimage\n",
+    "\n",
+    "class SigmaEstimateCallback(callbacks.Callback):\n",
+    "    \"\"\"\n",
+    "    Properties\n",
+    "    ----------\n",
+    "    save_values: list of floats \n",
+    "      The saved sigma calculation, one per iteration\n",
+    "      \n",
+    "    \"\"\"\n",
+    "    \n",
+    "    def __init__(self):\n",
+    "\n",
+    "        self.save_values=[]\n",
+    "\n",
+    "    def __call__(self, algorithm):\n",
+    "        self.save_values.append(skimage.restoration.estimate_sigma(algorithm.get_output().as_array()))\n",
+    "\n",
+    "mycallback_FISTA_TV_alpha_01= SigmaEstimateCallback()\n",
+    "algo1=FISTA(initial=ig.allocate(0), f=F, g=0.1*TotalVariation(lower=0), update_objective_interval=10)         \n",
+    "algo1.run(500, callbacks=[mycallback_FISTA_TV_alpha_01])\n",
+    "\n",
+    "        \n",
+    "mycallback_FISTA_TV_alpha_1= SigmaEstimateCallback()\n",
+    "algo2=FISTA(initial=ig.allocate(0), f=F, g=1*TotalVariation(lower=0), update_objective_interval=10)         \n",
+    "algo2.run(500, callbacks=[mycallback_FISTA_TV_alpha_1])\n",
+    "\n",
+    "\n",
+    "show2D([ground_truth, algo1.get_output(), algo2.get_output()], title=['ground_truth', 'FISTA_TV_alpha_01', 'FISTA_TV_alpha_1'], num_cols=3)\n",
+    "show2D([absorption, A.direct(algo1.get_output())-absorption, A.direct(algo2.get_output())-absorption], title=['ground_truth', 'Data error FISTA_TV_alpha_01', 'Data error FISTA_TV_alpha_1'], fix_range=[[0,3], [-0.02, 0.02], [-0.02, 0.02]], cmap=['gray', 'seismic', 'seismic'], num_cols=3)\n",
+    "plt.plot(range(10,501), mycallback_FISTA_TV_alpha_01.save_values[10:], label='FISTA TV alpha=0.1 ')\n",
+    "plt.plot(range(10, 501), mycallback_FISTA_TV_alpha_1.save_values[10:], label='FISTA TV alpha=1.0 ')\n",
+    "plt.ylabel('Noise Estimate')\n",
+    "plt.xlabel('Iteration')\n",
+    "plt.legend()\n",
+    "                                   \n",
+    "                            "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see with a larger regularisation parameter, the resulting image is less noisy. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Image metric callbacks (complex custom callback example) \n",
+    "We now move on to some more complex callbacks. In this callback, a dictionary of metrics are calculated and printed each `print_interval`. "
```

Comment:
might be nice to link to the CIL docs in case they want to learn any more about the metrics